src/core/lib/channel/channelz.cc

```diff
@@ -139,8 +128,17 @@ char* ChannelNode::RenderJSON() {
                                            GRPC_JSON_OBJECT, false);
   json = data;
   json_iterator = nullptr;
+
+  // query the stack below for relevant information.
+  grpc_channel_info channel_info;
+  memset(&channel_info, 0, sizeof(channel_info));
+  if (channel_ != nullptr) {
+    grpc_channel_get_info(channel_, &channel_info);
+  }
+
   // create and fill the connectivity state child.
-  grpc_connectivity_state connectivity_state = GetConnectivityState();
+  grpc_connectivity_state connectivity_state =
+      (channel_ != nullptr) ? channel_info.state : GRPC_CHANNEL_SHUTDOWN;
```

Comment:
It doesn't look like you've actually changed any filter to set `channel_info.state`, which means that it will always return the default value.